### PR TITLE
Sw dspdc 1063 transform st columns

### PR DESCRIPTION
**Pull request by snwessel**
Adding the study status columns to the dog aging transformation (just 5 columns). I was using this as an excuse to also:
- get rid of the massive HlesDog constructor, and use the new init method instead
- set up a test class for the dog transformations

@danxmoran (after the weekend) let me know if there is something else you had in mind for how to test these transformations. I would be happy to change, especially if something else would be easier in the long run.

### CODE REVIEW

File: schema/src/main/jade-tables/hles_dog.table.json [hunk 1]
```diff
@@ -17,6 +17,26 @@
         }
       ]
     },
+    {
+      "name": "st_vip_or_staff",
+      "datatype": "string"
```
Comment (snwessel) [inline schema/src/main/jade-tables/hles_dog.table.json:22]: I think I might have been getting ahead of the updates to the mapping doc... I'm not sure if this column should actually be an integer.

File: hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/DogTransformations.scala [hunk 2]
```diff
@@ -9,76 +9,14 @@ object DogTransformations {
 
   /** Parse all dog-related fields out of a raw RedCap record. */
   def mapDog(rawRecord: RawRecord): HlesDog = {
-    val dogBase = HlesDog(
+    val dogBase = HlesDog.init(
       dogId = rawRecord.id,
       // FIXME: Once DAP figures out a name for a dedicated owner ID, use that.
-      ownerId = rawRecord.id,
```
Comment (danxmoran) [inline hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/DogTransformations.scala:15]: Glorious

File: hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/RawRecord.scala [hunk 3]
```diff
@@ -43,6 +46,10 @@ case class RawRecord(id: Long, fields: Map[String, Array[String]]) {
   /** Get the singleton value for an attribute in this record if one exists, parsed as a boolean. */
   def getOptionalBoolean(field: String): Option[Boolean] = getOptional(field).map(_ == "1")
 
+  /** Get the singleton value for an attribute in this record if one exists, parsed as a date. */
+  def getOptionalDate(field: String): Option[LocalDate] =
+    getOptional(field).map(LocalDate.parse(_, DateTimeFormatter.ofPattern("MM-dd-yyyy")))
```
Comment (danxmoran) [inline hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/RawRecord.scala:51]: Need to check some example data to be sure this date format will actually work.

File: hle-survey/transformation/src/test/scala/org/broadinstitute/monster/dap/DogTransformationsSpec.scala [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import java.time.LocalDate
+
+class DogTransformationsSpec extends AnyFlatSpec with Matchers {
+  behavior of "DogTransformations"
+
+  // TODO add tests for dog demographics transformation
+
+  it should "correctly map study status fields" in {
+    val exampleDogFields = Map[String, Array[String]](
+      "dd_us_born" -> Array("1"), // a required field
```
Comment (danxmoran) [inline hle-survey/transformation/src/test/scala/org/broadinstitute/monster/dap/DogTransformationsSpec.scala:14]: Why is this required here?

File: schema/src/main/jade-tables/hles_dog.table.json (hunk 1 above)
Comment (danxmoran) [reply schema/src/main/jade-tables/hles_dog.table.json:22]: Yes, you got ahead of me! This one should be an int, the rest should be ok as-is.

File: hle-survey/transformation/src/test/scala/org/broadinstitute/monster/dap/DogTransformationsSpec.scala (hunk 4 above)
Comment (snwessel) [reply hle-survey/transformation/src/test/scala/org/broadinstitute/monster/dap/DogTransformationsSpec.scala:14]: It looks like one of the transformation methods uses getRequired for that field without being wrapped in a conditional. I had assumed it was intentional but maybe not? (the getRequired call is [here](https://github.com/DataBiosphere/dog-aging-ingest/blob/efc2a9f8f7acb2a3b34901412f1b23911478b831/hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/DogTransformations.scala#L257)).

Comment (danxmoran) [reply hle-survey/transformation/src/test/scala/org/broadinstitute/monster/dap/DogTransformationsSpec.scala:14]: Ah! Would change this test to call `mapStudyStatus` instead, so we can remove this.

File: hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/RawRecord.scala (hunk 3 above)
Comment (danxmoran) [reply hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/RawRecord.scala:54]: Weirdness... I can't actually find any examples of date files in extraction outputs. It's been a few weeks since I ran it so things might have updated in RedCap, @snwessel have you seen any examples IRL?

**Comment by snwessel** [comment]
@danxmoran no I haven't been able to find an actual example of a date yet either (although I only have a subset of the data). I was basing that format off of the redcap field description, quoted here:
```
"select_choices_or_calculations":"if(((datediff(\"02-29-2000\",[dd_form_date],\"d\",\"mdy\",true) - ...
```
